pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -687,5 +688,55 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
             arr[k] *= inv_norm;
         }
     }
+
+    /**
+     * @brief Prepares a single computational basis state.
+     *
+     * @param index Index of the target element.
+     */
+    void setBasisState(const std::size_t index) {
+        auto length = this->getLength();
+        PL_ABORT_IF(index > length - 1, "Invalid index");
+
+        auto *arr = this->getData();
+        for (size_t k = 0; k < length; k++) {
+            arr[k] = {0.0, 0.0};
+        }
```

Comment:
Can we keep the previous implementation? 
```suggestion
        std::fill(arr, arr + length, 0.0);
```